core/host_api/impl/crypto_extension.cpp

```diff
@@ -281,9 +286,10 @@ namespace kagome::host_api {
 
     auto sign = ed25519_provider_->sign(key_pair.value(), msg_buffer);
     if (!sign) {
-      logger_->error("failed to sign message, error = {}",
-                     sign.error().message());
-      std::terminate();
+      auto msg = fmt::format("failed to sign message, error = {}",
+                             sign.error().message());
+      logger_->error(msg);
+      throw std::runtime_error(msg);
```

Comment:
This piece of code occurs in so many places it could be extracted to a function.

Reply:
Point taken.